### PR TITLE
Discussion post endpoint update and image fix

### PR DESCRIPTION
**Pull request by askabderon**
This PR yields the changes below:

1. POST image endpoint is only returning the id of the image as JSON
```json
{
    "id": 109
}
```
2. GET discussionForum endpoint is updated to GET discussionPost
3. GET discussion/{id} endpoint is updated to GET discussionPost/{id} 
4. DELETE discussion/{id} endpoint is updated to DELETE discussionPost/{id} 
5.  POST discussion endpoint is updated to POST discussionPost

I updated the discussion endpoints to be consistent with Event and ArtItem endpoint structure.

### CODE REVIEW

File: backend/src/main/kotlin/com/group7/artshare/controller/DiscussionPostController.kt [hunk 1]
```diff
@@ -52,6 +52,10 @@ class DiscussionPostController (
             }
         }
     }
+    @GetMapping()
+    fun getRecommendedEventsGeneric(): List<DiscussionPost> {
```
Comment (sabrimete) [inline backend/src/main/kotlin/com/group7/artshare/controller/DiscussionPostController.kt:56]: getRecommendedEventsGeneric should be renamed.

**Review by sabrimete — CHANGES_REQUESTED** [review]
Small typo should be fixed

**Review by sabrimete — APPROVED** [review]
LGTM

**Comment by askabderon** [comment]
> Small typo should be fixed

Updated with the latest commit

**Pull request merged**